File: Python/1787. 【认证试题】最长元音子串的长度.py
'''
 Copyright (c) Huawei Technologies Co., Ltd. 2019-2020. All rights reserved.
 Description: 考生实现代码
 Note: 缺省代码仅供参考，可自行决定使用、修改或删除
'''


class Solution:
    def longest_vowel_length(self, in_str):
        vowel = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
        res = 0
        l = 0
        for s in in_str:
            if s in vowel:
                l += 1
            else:
                res = max(res, l)
                l = 0
        
        return max(res, l)


if __name__ == "__main__":
    in_str = input().strip()
    function = Solution()
    result = function.longest_vowel_length(in_str)
    print(result)

File: C++/1787. 【认证试题】最长元音子串的长度.cpp
/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2019-2020. All rights reserved.
 * Description: 考生实现代码
 * Note: 缺省代码仅供参考，可自行决定使用、修改或删除
 */
#include <iostream>
#include <string>
#include <set>
using namespace std;
int GetLongestVowelStringLength(const string& input)
{
    set<int> vowel {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
    int l = 0;
    int res = 0;
    for(auto s:input) {
        if(vowel.count(s)) {
            l++;
        } else {
            res = max(res, l);
            l = 0;
        }
    }
    return max(res, l);
}

int main()
{
    string input;
    cin >> input;

    cout << GetLongestVowelStringLength(input) << endl;
    return 0;
}
